tests/async_tests/test_interfaces.py

```diff
@@ -466,3 +467,34 @@ async def test_cannot_connect_uds(backend: str) -> None:
     async with httpcore.AsyncConnectionPool(backend=backend, uds=uds) as http:
         with pytest.raises(httpcore.ConnectError):
             await http.arequest(method, url)
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 7), reason="Hypercorn doesn't support python < 3.7"
+)
+@pytest.mark.anyio
+async def test_connection_timeout_tcp(backend: str, slow_tcp_server: Server) -> None:
+    method = b"GET"
+    url = (b"https", *slow_tcp_server.netloc, b"/delay/5")
+    headers = [slow_tcp_server.host_header]
+    ext = {"timeout": {"connect": 0.1}}
+
+    async with httpcore.AsyncConnectionPool(backend=backend) as http:
+        with pytest.raises(httpcore.TimeoutException):
+            await http.arequest(method, url, headers, ext=ext)
+
+
+@pytest.mark.skipif(
+    sys.version_info[:2] < (3, 7), reason="Hypercorn doesn't support python < 3.7"
+)
+@pytest.mark.anyio
+async def test_connection_timeout_uds(backend: str, slow_uds_server: Server) -> None:
+    uds = slow_uds_server.uds
+    method = b"GET"
+    url = (b"https", b"localhost", None, b"/delay/5")
```

Comment:
Currently this `/delay/5` path isn't doing anything special, but I agree it's good to be able to specify it from the test. In my other comment I suggest parsing the amount of seconds from the requested URL path, so that requesting this `/delay/5` would effectively make the test server respond after 5 seconds.